subprojects/model-core/src/test/groovy/org/gradle/internal/extensibility/ExtensibleDynamicObjectTest.java

```diff
@@ -536,6 +536,17 @@ public void canInvokeClosurePropertyAsAMethod() {
         assertThat(bean.invokeMethod("someMethod", "Param"), equalTo((Object) "param"));
     }
 
+    @Test
+    public void canInvokeClosureLikePropertyAsAMethod() {
+        Bean bean = new Bean();
+        bean.defineProperty("someMethod", new Object() {
+            public String call(String param) {
+                return param.toLowerCase();
+            }
+        });
+        assertThat(bean.invokeMethod("someMethod", "Param"), equalTo("param"));
+    }
+
```

Comment:
Like @jvandort said, let's put this into an integration test using a build script.

I think we can put it in model-core integTest as a new test class.  It can be something like (all in the build script file itself):

- Create a domain object container with `ObjectFactory.domainObjectContainer`
- Create an interface with a `void call(String)` method
- Create an element class that returns the interface from a getter

You should be able to create/register an element in the container and show that each of these work:

```
container.element.foo "bar"
container.element {
   foo "bar"
}
configure(container) {
   element {
      foo "bar"
   }
}
```

Reply:
Added, though it's slightly different because our DSL didn't seem to mix in everything it normally would.